### PR TITLE
Basic dialogue choice state view

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

File: go/base/static/js/src/campaign/dialogue/states/states.js [hunk 1]
```diff
@@ -31,11 +35,27 @@
       return this;
     },
 
+    detach: function() {
+      this.$el.detach();
+      return this;
+    },
+
     render: function() {
+      var data = {model: this.state.model.toJSON()};
```
Comment (hodgestar) [inline go/base/static/js/src/campaign/dialogue/states/states.js:44]: Why do the templates need a JSON string of the model?

File: go/base/static/js/src/campaign/dialogue/states/states.js [hunk 2]
```diff
@@ -43,19 +63,93 @@
...
+    onTypeChange: function(e) {
+      this.state.reset($(e.target).val());
+    },
+
+    save: function() { return this; },
+
+    cancel: function() {
+      var model = this.state.model;
+      model.clear();
+      model.set(this.modelBackup);
```
Comment (hodgestar) [inline go/base/static/js/src/campaign/dialogue/states/states.js:118]: Maybe we should check that `this.modelBackup` is set to something before doing the `model.clear()`?

File: go/base/static/js/test/tests/campaign/dialogue/states/choice.test.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+          text: "What is your favourite colour?",
+          ordinal: 0,
+          entry_endpoint: {uuid: "endpoint0"},
+          choice_endpoints: [
+            {value: "value1", label: "Red", uuid: "endpoint1"},
+            {value: "value2", label: "Blue", uuid: "endpoint2"}]
+        });
+
+        editMode.$('.text').text('What is your favourite dinosaur?');
+        editMode.$('.choice input').eq(1).val('Diplodocus');
```
Comment (hodgestar) [inline go/base/static/js/test/tests/campaign/dialogue/states/choice.test.js:51]: Cool dinosaur.

File: go/base/static/js/src/campaign/dialogue/states/states.js (hunk 2 above)
Comment (justinvdm) [inline go/base/static/js/src/campaign/dialogue/states/states.js:118]: Very good point! Thanks :)

File: go/base/static/js/src/campaign/dialogue/states/states.js (hunk 1 above)
Comment (hodgestar) [inline go/base/static/js/src/campaign/dialogue/states/states.js:44]: P.S. The `.toJSON()` part is the bit I'm surprised by.

**Comment by hodgestar** [comment]
Other than the query about the `toJSON()` part, I'm :+1:.

File: go/base/static/js/src/campaign/dialogue/states/states.js (hunk 1 above)
Comment (justinvdm) [inline go/base/static/js/src/campaign/dialogue/states/states.js:44]: `toJSON()` returns a json object. I do this to make accessing things in the template simpler, but its probably not a good enough reason. I'll change this to just use the model.

File: go/base/static/js/test/tests/campaign/dialogue/states/choice.test.js (hunk 3 above)
Comment (justinvdm) [inline go/base/static/js/test/tests/campaign/dialogue/states/choice.test.js:51]: Hehe, thanks! :)

File: go/base/static/js/src/campaign/dialogue/states/states.js (hunk 1 above)
Comment (hodgestar) [inline go/base/static/js/src/campaign/dialogue/states/states.js:44]: Ah! /me reads http://backbonejs.org/#Model-toJSON.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**